cli/compile/compile.go

```diff
@@ -83,6 +84,10 @@ func NewCommand() *cobra.Command {
 	}
 
 	command.Flags().StringVarP(&fqbn, "fqbn", "b", "", tr("Fully Qualified Board Name, e.g.: arduino:avr:uno"))
+	command.MarkFlagRequired("fqbn")
```

Comment:
This flag is not required due to the ability to define it in the sketch metadata via `arduino-cli board attach`:
```
$ ./arduino-cli version
arduino-cli.exe alpha Version: 0.19.0 Commit: 56419ecd Date: 2021-09-02T14:47:35Z

$ ./arduino-cli sketch new /tmp/FooSketch
Sketch created in: C:\Users\per\AppData\Local\Temp\FooSketch

$ ./arduino-cli board attach arduino:avr:uno /tmp/FooSketch
Selected fqbn: arduino:avr:uno

$ ./arduino-cli compile /tmp/FooSketch
Sketch uses 444 bytes (1%) of program storage space. Maximum is 32256 bytes.
Global variables use 9 bytes (0%) of dynamic memory, leaving 2039 bytes for local variables. Maximum is 2048 bytes.
```

Reply:
you are right, I thought about it later... I just wanted to experiment with `MarkFlagRequired` function. We can use this somewhere else:
basically if you type `completion <tab><tab>` the completion suggest at first only the required flags, so the user is prompted to use them.

Reply:
BTW I reverted it :smile: